com.wso2telco.dep.gw.smsapi/smsapigw-synapse/src/main/synapse-config/sequences/sendSmsOutSeq.xml

```diff
@@ -1,29 +1,45 @@
+    <filter source="boolean(get-property('operatorRequestId'))" regex="false">
+        <then>
+            <log level="custom">
+                <property name="ERROR" value="Unable extract operator request id (plugin_requestid) from response"/>
+            </log>
+        </then>
+    </filter>
 
-            // set request id sent from operator side to message context
-            mc.setProperty("SEND_SMS_OPERATOR_REQUEST_ID", operatorRequestId);
-            var resourceUrl = smsRetrieveResourceUrlPrefix + "/" + encodedSenderAddress + "/requests/" + requestId;
-            if (mc.getProperty("originalNotifyURL")) {
-            payload.outboundSMSMessageRequest.receiptRequest.notifyURL = mc.getProperty("originalNotifyURL");
-            }
-            payload.outboundSMSMessageRequest.resourceURL = resourceUrl;
-            payload.outboundSMSMessageRequest.deliveryInfoList.resourceURL = resourceUrl + "/deliveryInfos";
-            mc.setPayloadJSON(payload);
-          '/>
+    <property name="SEND_SMS_OPERATOR_REQUEST_ID" expression="get-property('operatorRequestId')" action="set" scope="default"/>
+
+    <property name="smsRetrieveResourceUrl" expression="fn:concat(syn:get-property('SEND_SMS_RESOURCE_URL_PREFIX'), '/requests/', syn:get-property('REQUEST_ID'))" scope="default" xmlns:fn="http://www.w3.org/2005/xpath-functions"/>
+    <property name="smsDeliveryInfoResourceUrl" expression="fn:concat(syn:get-property('smsRetrieveResourceUrl'), '/deliveryInfos')" scope="default" xmlns:fn="http://www.w3.org/2005/xpath-functions"/>
+
+    <filter source="boolean(get-property('originalNotifyURL'))" regex="true">
+        <then>
+            <class name="com.wso2telco.dep.common.mediation.UtilMediator">
+                <property name="propertyValue" value="singlePropertyReplacement"/>
+                <property name="propertyName" value="notifyURL"/>
+                <property name="propertyPath" value="payload.outboundSMSMessageRequest.receiptRequest.notifyURL"/>
+                <property name="msgContextProperty" value="originalNotifyURL"/>
+            </class>
+        </then>
+    </filter>
+
+    <class name="com.wso2telco.dep.common.mediation.UtilMediator">
+        <property name="propertyValue" value="singlePropertyReplacement"/>
+        <property name="propertyName" value="resourceURL"/>
+        <property name="propertyPath" value="payload.outboundSMSMessageRequest.resourceURL"/>
+        <property name="msgContextProperty" value="smsRetrieveResourceUrl"/>
     </class>
+
+    <class name="com.wso2telco.dep.common.mediation.UtilMediator">
+        <property name="propertyValue" value="singlePropertyReplacement"/>
+        <property name="propertyName" value="resourceURL"/>
+        <property name="propertyPath" value="payload.outboundSMSMessageRequest.deliveryInfoList.resourceURL"/>
+        <property name="msgContextProperty" value="smsDeliveryInfoResourceUrl"/>
+    </class>
+
```

Comment:
Better to find a way of updating multiple SinglePropertyReplacement at once that will prevent the calling UtilMediator multiple times.